### DISCUSSION TITLE
Test out a configuration without opening a new terminal

### DISCUSSION THREAD

**Discussion by ittayd** [wezterm/wezterm#2135] [Q&A]
My understanding is that if I want to test out a new configuration (I'm especially interested in key bindings) I need to change .wezterm.lua then launch a new terminal. Is there a way to test it out in the current terminal? 

Similarly, is there a way to run actions directly?

**Comment by ittayd** [comment]
Continuing this, is there a way to view the current configuration? E.g. I can't tell from the documentation what the quick select patterns are, how can I see in the terminal?

**Reply by wez** [reply]
When you save the configuration file it will be reloaded and apply immediately.
You can force the config to reload by pressing `CTRL-SHIFT-R`
You can see the current config by entering the debug overly (`CTRL-SHIFT-L`) and then typing `window:effective_config()`.  Press ESC to exit the debug overlay.

**Reply by ittayd** [reply]
Thank you. Is there a way to just type in the configuration in the terminal rather than edit the file, which then means it is loaded by all terminals?
Also, note that window:effective_config() doesn't output the default key bindings. It would be also useful if it output what configuration files were loaded.

**Reply by ittayd** [reply]
So for example, I was trying to make changes to wezterm.lua and had an error (not sure even what), then ctrl-shift-l stopped working in all terminals and I had to move the lua file to get things working again.

**Reply by wez** [reply]
What I do is have vim open with the config inside wezterm, then simply save the file and hit CTRL-SHIFT-R as needed.

**Reply by wez** [reply]
The default key assignments are documented here: https://wezfurlong.org/wezterm/config/default-keys.html
